### PR TITLE
Economy: Beautify Ladders

### PR DESCRIPTION
**Pull request by Lord-Haji**
CSS Patch: https://hastebin.com/powopuxose.css
This Beautifies the Ladders. Also The Universal Code for other ladders is now Ready. If we have any other ladder anytime soon. ex: - tourladder, expladder

# `Screen Shots`
## Light Mode Normal
cap1
## Light Mode PM
cap2
## Dark Mode Normal
cap3
## Dark Mode PM
cap4

### CODE REVIEW

**Comment by ghost** [comment]
Oooh I like it

File: spacialgaze-plugins/economy.js [hunk 1]
```diff
@@ -235,31 +273,14 @@ exports.commands = {
...
-		let obj = Db.currency.keys().map(function (name) {return {name: name, amount: Db.currency.get(name)};});
-		let results = obj.sort(function (a, b) {
-			return b.amount - a.amount;
+		let keys = Db.currency.keys().map(function (name) {
+			return {name: name, money: Db.currency.get(name)};
 		});
-		showResults(results.slice(0, target));
+		if (!keys.length) return this.sendReplyBox("Money ladder is empty.");
+		keys.sort(function (a, b) { return b.money - a.money; });
+		this.sendReplyBox(rankLadder('Richest Users', 'Money', keys.slice(0, target), 'money') + '</div>');
```
Comment (HoeenCoder) [inline spacialgaze-plugins/economy.js]: Use the `currencyPlural` variable instead of `Money`

File: spacialgaze-plugins/economy.js [hunk 2]
```diff
@@ -273,7 +294,7 @@ exports.commands = {
 		user.customSymbol = target;
 		user.updateIdentity();
 		user.canCustomSymbol = false;
-		this.sendReply('Your symbol is now ' + target + '. It will be saved until you log off for more than an hour, or the server restarts. You can remove it with /resetsymbol');
+		this.sendReply('Your Symbol is now ' + target + '. It will be saved until you log off for more than an hour, or the server restarts. You can remove it with /resetsymbol');
```
Comment (HoeenCoder) [inline spacialgaze-plugins/economy.js]: This was correct before.

Comment (Lord-Haji) [reply spacialgaze-plugins/economy.js]: Should I revert this?

**Comment by Mystifi** [comment]
The colours (specifically gold and bronze) need to be less saturated; in your example image, it's hard to see Insist's name.

**Comment by Mystifi** [comment]
Also, I'm not too sure if we really need the user's global auth stated in the command, but I'm open to reasons why

File: spacialgaze-plugins/economy.js (hunk 2 above)
Comment (HoeenCoder) [reply spacialgaze-plugins/economy.js]: yeah change this back to `symbol`

**Comment by Lord-Haji** [comment]
It's a parameter added to nameColor

**Comment by HoeenCoder** [comment]
an optional paramater ofc, idc either way

**Comment by ghost** [comment]
Its just a matter of preference atm. We DONT need it, but its nice anyways

**Comment by Mystifi** [comment]
I don't think "it's a parameter" really answers my question. I mean, ranks don't need to be there unless you had something where it sorted the richestusers by rank? That could be a idea.

**Comment by Mystifi** [comment]
Also, I'd like the bronze to be just a little lighter; as I said earlier, it's a bit hard to see Insist's username.

**Comment by HoeenCoder** [comment]
Ok so things to change:

- Dont show rank, its unneeded.

- Edit the colors slightly so its easier to see names like Insist's

**Comment by Lord-Haji** [comment]
@HoeenCoder @Mystifi removed usegroup display, And Bronze is a bit light now

**Comment by DeathlyPlays** [comment]
Ooooh I like this tbh :o

**Comment by ghost** [comment]
> And Bronze is a bit light now

Can we see a screenshot of the new bronze color?

File: spacialgaze-plugins/economy.js [hunk 3]
```diff
@@ -235,31 +273,16 @@ exports.commands = {
...
-				count++;
-			}
-			self.sendReplyBox(output);
-			if (room) room.update();
-		}
-		let obj = Db.currency.keys().map(function (name) {return {name: name, amount: Db.currency.get(name)};});
-		let results = obj.sort(function (a, b) {
-			return b.amount - a.amount;
+		let keys = Db.currency.keys().map(function (name) {
+			return {name: name, money: Db.currency.get(name)};
```
Comment (Mystifi) [inline spacialgaze-plugins/economy.js]: The callback can be simplified to `name => ({name, money: Db.currency.get(name)})`

Comment (Lord-Haji) [reply spacialgaze-plugins/economy.js]: Sorry I didn't get you

**Comment by Lord-Haji** [comment]
scrnsht

Bump @Mystifi @HoeenCoder

**Pull request merged**